CMakeLists.txt

```diff
@@ -0,0 +1,474 @@
+cmake_minimum_required (VERSION 2.8)
+project (logcabin)
+set (LOGCABIN_VERSION_MAJOR 1)
+set (LOGCABIN_VERSION_MINOR 2)
+
+include_directories(BEFORE .)
+include_directories(BEFORE ./include)
+include_directories(BEFORE ./build)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-strict-overflow -fPIC")
+
+# TODO: accept user input
+set(CMAKE_PROTO_CXX_FLAGS " ")
+set(CMAKE_GTEST_CXX_FLAGS " ")
+
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -DDEBUG")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O2 -DNDEBUG")
+
+file(MAKE_DIRECTORY build)
+
+cmake_policy(SET CMP0015 NEW)
+link_directories(build)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/build)
+
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    message("using clang")
```

Comment:
I think you can safely get rid of this now, since it's shown in the default cmake output anyway. Same with "using gcc".